docs/source/modules/api.rst

```diff
@@ -1,41 +1,18 @@
 api
 ===
 
-.. toctree::
-   :maxdepth: 1
-   :hidden:
+.. autosummary::
+    :nosignatures:
+    :toctree: STUBDIR
+    :template: autosummary/module.rst
 
-   api/basic
-   api/discretization
-   api/equation
-   api/essential_bc
-   api/expr
-   api/feec
-   api/fem
-   api/glt
-   api/grid
-   api/postprocessing
-
-.. list-table::
-   :widths: 20 80
-
-   * - :doc:`basic <api/basic>`
-     - .. automodule:: psydac.api.basic
-   * - :doc:`discretization <api/discretization>`
-     - .. automodule:: psydac.api.discretization
-   * - :doc:`equation <api/equation>`
-     - .. automodule:: psydac.api.equation
-   * - :doc:`essential_bc <api/essential_bc>`
-     - .. automodule:: psydac.api.essential_bc
-   * - :doc:`expr <api/expr>`
-     - .. automodule:: psydac.api.expr
-   * - :doc:`feec <api/feec>`
-     - .. automodule:: psydac.api.feec
-   * - :doc:`fem <api/fem>`
-     - .. automodule:: psydac.api.fem
-   * - :doc:`glt <api/glt>`
-     - .. automodule:: psydac.api.glt
-   * - :doc:`grid <api/grid>`
-     - .. automodule:: psydac.api.grid
-   * - :doc:`postprocessing <api/postprocessing>`
-     - .. automodule:: psydac.api.postprocessing
+    psydac.api.basic
+    psydac.api.discretization
+    psydac.api.equation
+    psydac.api.essential_bc
+    psydac.api.expr
+    psydac.api.feec
+    psydac.api.fem
+    psydac.api.glt
+    psydac.api.grid
+    psydac.api.postprocessing
```

Comment:
Have you tried to use the `currentmodule` directive?
```rst
.. currentmodule: psydac.api
.. autosummary::
    :nosignatures:
    :toctree: STUBDIR
    :template: autosummary/module.rst

    basic
    discretization
    equation
    essential_bc
    expr
    feec
    fem
    glt
    grid
    postprocessing
````

Reply:
That's a really good point, I had forgotten about the `.. currentmodule::` directive!
I propose writing 
```
.. currentmodule: psydac
.. autosummary::
    :nosignatures:
    :toctree: STUBDIR
    :template: autosummary/module.rst

    api.basic
    api.discretization
    api.equation
    api.essential_bc
    api.expr
    api.feec
    api.fem
    api.glt
    api.grid
    api.postprocessing
```

As we have several modules named `basic` for example. This way, they will appear everywhere as `submodule.basic`, both in the Navigation sidebar and on the module overview page.